### PR TITLE
add role by a new button in role section

### PR DESCRIPTION
**Pull request by SAURAABHPATEL**
Targets #2423 
add role by a new button in role section.

### Description
 you have to add new role button in role section. whenever anybody hit that button then it should be open a form and that form requires all input column of database roles and after filling that data of form it will save in roles column as well as in database table.

### Checklist:

- [ ] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/User/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -66,4 +68,29 @@ public function destroy(User $user)
...
+            'name' => 'required|unique:roles,name',
+        ]);
+        $role = new Role();
+        $role->name = $request->name;
+        $role->label = $request->label;
+        $role->guard_name = $request->guard_name;
+        $role->description = $request->description;
+        $role->save();
+
+        return redirect()->back();
```
Comment (gautam071) [inline Modules/User/Http/Controllers/UserController.php:84]: use service and request files

File: Modules/User/Resources/assets/js/components/RoleListing.vue [hunk 2]
```diff
@@ -59,6 +64,12 @@ export default {
 
 		onRolePressionsUpdated: function(selectedPermissions) {
 			Vue.set(this.selectedRole, "permissions", selectedPermissions);
+		},
+
+		removeRole: async function(index) {
+			let id = this.allRoles[index]["id"];
+			let route = `DeleteRoles/${id}`;
+			this.allRoles.splice(index, 1);
 		}
```
Comment (gautam071) [inline Modules/User/Resources/assets/js/components/RoleListing.vue:73]: why this is needed?

File: Modules/User/Resources/assets/js/components/RoleListing.vue [hunk 3]
```diff
@@ -36,7 +40,8 @@ export default {
 			currentUserIndex: 0,
 			roleInputs: [],
 			allRoles: this.roles,
-			selectedRole:{}
+			selectedRole:{},
+			// rolelocal: this.roles,
```
Comment (gautam071) [inline Modules/User/Resources/assets/js/components/RoleListing.vue:44]: remove the comment code

File: Modules/User/Resources/assets/js/components/RoleListing.vue [hunk 4]
```diff
@@ -14,6 +15,9 @@
 						<div class="mb-1">{{ role.label }}</div>
 						<div class="text-muted fz-14">{{ role.description }}</div>
 					</td>
+					<td>
+						<button v-if="role.id != '1'" v-on:click="removeRole(index)" class="btn btn-danger" id="disablebtn">Delete</button>
```
Comment (gautam071) [inline Modules/User/Resources/assets/js/components/RoleListing.vue:19]: use by `guard_name`

File: Modules/User/Resources/assets/js/components/RoleListing.vue [hunk 5]
```diff
@@ -4,6 +4,7 @@
 			<thead class="thead-dark">
 				<tr>
 					<th>Role</th>
+					<th>Correction</th>
```
Comment (gautam071) [inline Modules/User/Resources/assets/js/components/RoleListing.vue:7]: What is this?

File: Modules/User/Resources/views/roles/index.blade.php [hunk 6]
```diff
@@ -4,12 +4,58 @@
 <div class="container" id="vueContainer">
     @includeWhen(session('success'), 'toast', ['message' => session('success')])
 	@include('user::layouts.navbar')
-	<h4>Roles management</h4>
+	<h4>Roles management </h4>
+	<div class="d-none alert alert-success " id="successMessage" role="alert">
+        <strong>Updated!</strong> Submitted successfully.
```
Comment (gautam071) [inline Modules/User/Resources/views/roles/index.blade.php:9]:
```suggestion
        <strong>Updated!</strong> the Role successfully.
```

File: Modules/User/Resources/views/roles/index.blade.php [hunk 7]
```diff
@@ -4,12 +4,58 @@
...
+        <button type="button" class="close" id="closeSuccessMessage" aria-label="Close">
+        </button>
+    </div>
+	<button type="button" class="btn btn-primary float-right mb-6" data-toggle="modal" data-target="#rolesModal">
+	ADD NEW ROLE
```
Comment (gautam071) [inline Modules/User/Resources/views/roles/index.blade.php:14]:
```suggestion
	Add New Role 
```

File: Modules/User/Resources/views/roles/index.blade.php [hunk 8]
```diff
@@ -4,12 +4,58 @@
 <div class="container" id="vueContainer">
     @includeWhen(session('success'), 'toast', ['message' => session('success')])
 	@include('user::layouts.navbar')
-	<h4>Roles management</h4>
+	<h4>Roles management </h4>
```
Comment (gautam071) [inline Modules/User/Resources/views/roles/index.blade.php:7]: check the formatting in the bklade file

File: Modules/User/Resources/views/roles/index.blade.php [hunk 9]
```diff
@@ -4,12 +4,58 @@
...
 		:roles="{{ json_encode($roles) }}"
 		:permissions = "{{ json_encode($permissions)  }}"
 		:update-route="{{ json_encode( route('permissions.module.index', ['module' => 'roles'])) }}"
 	/>
+</div>
+	<div class="modal fade" id="rolesModal" tabindex="-1" role="dialog" aria-labelledby="rolesModalLabel" aria-hidden="true">
+		<div class="modal-dialog" role="document">
+			<div class="modal-content">
+				<div class="modal-header">
+					<h5 class="modal-title" id="rolesModalLabel">ADD NEW ROLE</h5>
```
Comment (gautam071) [inline Modules/User/Resources/views/roles/index.blade.php:27]: Same here

File: Modules/User/Routes/web.php [hunk 10]
```diff
@@ -17,7 +17,9 @@
     Route::get('roles', 'RolesController@index')->name('user.role-index');
     Route::delete('{user}/delete', 'UserController@destroy')->name('user.delete');
     Route::put('update-roles', 'UserController@updateUserRoles')->name('user.update-roles');
-
+    Route::get('/showForm', 'UserController@showForm')->name('user.show-form');
```
Comment (gautam071) [inline Modules/User/Routes/web.php:20]: why a route for modal?

File: database/Seeders/RolesTableSeeder.php [hunk 11]
```diff
@@ -50,7 +50,7 @@ public function run()
                'label'=> 'Intern',
                'guard_name' => 'web',
                'description' => 'Someone who is in training',
-            ]
+            ],
```
Comment (gautam071) [inline database/Seeders/RolesTableSeeder.php:53]:
```suggestion
            ]
```

File: resources/js/app.js [hunk 12]
```diff
@@ -1197,6 +1197,35 @@ if (document.getElementById("roles_permission_table")) {
...
+				$("#successMessage").fadeToggle(3000);
+			},
+			error: function (response) {
+				if (response.responseJSON.errors.name) {
+					let text = response.responseJSON.errors.name[0];
+					$("#roleserror").html(text).removeClass("d-none");
+					return false;
+				}
+			},
+		});
```
Comment (gautam071) [inline resources/js/app.js:1225]: What is the purpose of JQuery

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: Modules/User/Resources/assets/js/components/RoleListing.vue (hunk 2 above)
Comment (saurabhpatel123456) [reply Modules/User/Resources/assets/js/components/RoleListing.vue:73]: for deleting the roles.

File: Modules/User/Resources/assets/js/components/RoleListing.vue (hunk 3 above)
Comment (saurabhpatel123456) [reply Modules/User/Resources/assets/js/components/RoleListing.vue:44]: ok

File: resources/js/app.js (hunk 12 above)
Comment (saurabhpatel123456) [reply resources/js/app.js:1225]: it is used for checking  the similar data in database and then give a message to the user.

File: Modules/User/Resources/assets/js/components/RoleListing.vue (hunk 4 above)
Comment (saurabhpatel123456) [reply Modules/User/Resources/assets/js/components/RoleListing.vue:19]: ok

File: Modules/User/Resources/assets/js/components/RoleListing.vue (hunk 5 above)
Comment (saurabhpatel123456) [reply Modules/User/Resources/assets/js/components/RoleListing.vue:7]: i renamed.

File: Modules/User/Routes/web.php (hunk 10 above)
Comment (saurabhpatel123456) [reply Modules/User/Routes/web.php:20]: sorry i forgot to remove this route, i made for the form.

File: Modules/User/Resources/assets/js/components/RoleListing.vue (hunk 5 above)
Comment (saurabhpatel123456) [reply Modules/User/Resources/assets/js/components/RoleListing.vue]: that is the title of delete button but now i have changed the name.

**Pull request closed without merging**